File: DotNet/CesiumLanguageWriter/Advanced/ICesiumHorizontalOriginValuePropertyWriter.cs
namespace CesiumLanguageWriter.Advanced
{
    /// <summary>
    /// A writer that can write a value as a horizontal origin.
    /// </summary>
    public interface ICesiumHorizontalOriginValuePropertyWriter : ICesiumPropertyWriter
    {
        /// <summary>
        /// Writes the value expressed as a horizontal origin.
        /// </summary>
        /// <param name="value">The horizontal origin.</param>
        void WriteHorizontalOrigin(CesiumHorizontalOrigin value);
    }
}

File: Java/CesiumLanguageWriter/translatedSrc/cesiumlanguagewriter/advanced/ICesiumHorizontalOriginValuePropertyWriter.java
package cesiumlanguagewriter.advanced;


import agi.foundation.compatibility.*;
import cesiumlanguagewriter.*;
import javax.annotation.Nonnull;

/**
 * A writer that can write a value as a horizontal origin.
 */
@SuppressWarnings({
    "unused",
    "deprecation",
    "serial"
})
public interface ICesiumHorizontalOriginValuePropertyWriter extends ICesiumPropertyWriter {
    /**
    * Writes the value expressed as a horizontal origin.
    * @param value The horizontal origin.
    */
    void writeHorizontalOrigin(@Nonnull CesiumHorizontalOrigin value);
}